### PR TITLE
Latest changes for Dimuon Vertices

### PR DESCRIPTION
**Pull request by kerstinlovisa**
Mostly been working on dimuon vertex selections in TTAlpsObjectsManager and TTAlpsHistogramFiller.

### CODE REVIEW

File: apps/ttalps_histogrammer.cpp [hunk 1]
```diff
@@ -7,26 +7,33 @@
...
-    fatal() << "or"<<endl;
-    fatal() << argv[0] << " config_path input_path output_path"<<endl;
+int main(int argc, char **argv) {
+  
+  auto args = make_unique<ArgsManager>(argc, argv);
+  // check if optional value "config" is present
+  if (!args->GetString("config").has_value()) {
+    fatal() << "No config file provided" << endl;
     exit(1);
   }
```
Comment (jniedzie) [inline apps/ttalps_histogrammer.cpp:21]: I was wondering if maybe we should have two sets of functions, something like`GetOptionalString` and `GetString`. The first one would work as it is now, but the second one would automatically print a fatal error and exit if the value was not found. Like this, we would avoid checking it every time on the "user's" side. If you think that makes sense, I can change that.

File: apps/ttalps_histogrammer.cpp [hunk 2]
```diff
@@ -7,26 +7,33 @@
...
+    config.SetInputPath(args->GetString("input_path").value());
+  }
+  if (args->GetString("output_path").has_value()) {
+    config.SetHistogramsOutputPath(args->GetString("output_path").value());
+  }
+  if (args->GetString("output_hists_path").has_value()) {
+    config.SetHistogramsOutputPath(args->GetString("output_hists_path").value());
+  }
+  if (args->GetString("output_trees_path").has_value()) {
+    config.SetTreesOutputPath(args->GetString("output_trees_path").value());
```
Comment (jniedzie) [inline apps/ttalps_histogrammer.cpp:36]: Following the comment above, we could also move the check of whether the value exists or is null to the config manager.

File: configs/ttalps_cross_sections.py [hunk 3]
```diff
@@ -24,8 +24,8 @@
   "ttWJets_TuneCP5_13TeV_madgraphMLM_pythia8": 0.4611,
   
   "ttHToMuMu_M125_TuneCP5_13TeV-powheg-pythia8": 0.5269,
-  "ttHTobb_ttToSemiLep_M125_TuneCP5_13TeV-powheg-pythia8": 0.5418,
-  "ttHToNonbb_M125_TuneCP5_13TeV-powheg-pythia8": 0.5638,
+  "ttHTobb_ttToSemiLep_M125_TuneCP5_13TeV-powheg-pythia8": 0.2953,
+  "ttHToNonbb_M125_TuneCP5_13TeV-powheg-pythia8": 0.2118,
```
Comment (jniedzie) [inline configs/ttalps_cross_sections.py:28]: Maybe just add a comment here to explain what exactly was done for these cross sections.

File: libs/user_extensions/src/TTAlpsEvent.cpp [hunk 4]
```diff
@@ -140,7 +140,7 @@ bool TTAlpsEvent::IsGoodMuonFromALP(int genMuonIndex) {
   // loop over pythia copies of the muon
   if (mother->IsMuon()) {
```
Comment (jniedzie) [inline libs/user_extensions/src/TTAlpsEvent.cpp:141]: Do we need this if statement?

File: libs/user_extensions/src/TTAlpsEvent.cpp [hunk 5]
```diff
@@ -140,7 +140,7 @@ bool TTAlpsEvent::IsGoodMuonFromALP(int genMuonIndex) {
...
     while(mother->IsMuon()) {
-      motherIndex = muon->GetMotherIndex();
+      motherIndex = mother->GetMotherIndex();
       if (motherIndex < 0) return false;
       mother = asNanoGenParticle(genParticles->at(motherIndex));
     }
```
Comment (jniedzie) [inline libs/user_extensions/src/TTAlpsEvent.cpp:146]: Can we make this a method of `NanoGenParticle`? I.e. implement `GetFirstCopy` that will return the first copy of a particle regardless of its PID?

File: utils/merge_nanoAOD_files.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
+import glob
```
Comment (jniedzie) [inline utils/merge_nanoAOD_files.py:1]: Is this based on the LbL script? Do you think we could make it more general and a part of tea?

File: libs/user_extensions/src/TTAlpsEvent.cpp (hunk 5 above)
Comment (kerstinlovisa) [reply libs/user_extensions/src/TTAlpsEvent.cpp:146]: I don't think I understand what you mean with which method, but we need the event to get the complete collection of genparticles to change the mothers

File: utils/merge_nanoAOD_files.py (hunk 6 above)
Comment (kerstinlovisa) [reply utils/merge_nanoAOD_files.py:1]: Yeah I guess, we also have merge_histograms in ttalps utils

File: apps/ttalps_histogrammer.cpp (hunk 2 above)
Comment (kerstinlovisa) [reply apps/ttalps_histogrammer.cpp:36]: Yeah I agree would be good to clean this up a bit

**Pull request merged**